04/src/academy/pocu/comp2500samples/w04/singletonlogger/Program.java

```diff
@@ -0,0 +1,27 @@
+package academy.pocu.comp2500samples.w04.singletonlogger;
+
+import java.io.IOException;
+
+public class Program {
+    public static void main(String[] args) {
+        try {
+            Logger logger = Logger.getInstance();
+            logger.logDebug("Printed if the minimum log level is DEBUG");
+            logger.logInformation("Printed if the minimum log level is INFORMATION or lower");
+            logger.logWarning("Printed if the minimum log level is WARNING or lower");
+            logger.logError("Printed if the minimum log level is ERROR or lower");
+            logger.logCritical("Printed if the minimum log level is CRITICAL or lower");
+
+            int x = 10;
+            int y = 20;
+
+            logger.logDebug("%d * %d = %d", x, y, x * y);
+            logger.logInformation("%d * %d = %d", x, y, x * y);
+            logger.logWarning("%d * %d = %d", x, y, x * y);
+            logger.logError("%d * %d = %d", x, y, x * y);
+            logger.logCritical("%d * %d = %d", x, y, x * y);
+        } catch (IOException e) {
+            e.printStackTrace();

```

Comment:
usually u print both ToString and stacktrace I assume. so do it.

And it's fine to catch Exception on top level (and that's prolly more desirable. Show exception and exit program

Reply:
You mean like this?

```
System.err.println(e.toString());
e.printStackTrace();
```

But e.toString() is the first line of e.printStackTrace()...

Reply:
is it? then ignore what i said